File: 1248-count-number-of-nice-subarrays/1248-count-number-of-nice-subarrays.java
class Solution {
    
    public int atmost(int[] arr,int k){
        int n=arr.length;
        int start=0,ans=0,odd=0;
        for (int i=0;i<n;i++){
            if(arr[i]%2==1) odd++;
            while(odd>k){
                if(arr[start]%2==1) odd--;
                start++;
            }
            ans+=i-start+1;
        }
        return ans;
    }
    public int numberOfSubarrays(int[] nums, int k) {
        return (atmost(nums,k)-atmost(nums,k-1));
    }
}

File: 1248-count-number-of-nice-subarrays/1248-count-number-of-nice-subarrays.py
class Solution:
    
    def countAtMost(self, nums, k):
        left = 0
        res = 0
        for right in range(len(nums)):
            if nums[right] % 2 != 0:
                k -= 1
            while k < 0:
                if nums[left] % 2 != 0:
                    k += 1
                left += 1
            res += (right - left + 1)
        return res
    def numberOfSubarrays(self, nums: List[int], k: int) -> int:
        return self.countAtMost(nums,k)-self.countAtMost(nums,k-1)